### PR TITLE
feat: implement the eth filter methods as todo!

### PR DESCRIPTION
**Pull request by aniketpr01**
Time spent on this PR: 2

Resolves: [397](https://github.com/kkrt-labs/kakarot-rpc/issues/397)

# Pull Request type
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?
we will be able to add functional logic for eth filter methods which are marked for todo as of now.

# Does this introduce a breaking change?

- [ ] Yes
- [✔️] No

### CODE REVIEW

**Review by Eikix — COMMENTED** [review]
Hey! Welcome! Thanks for your PR!
Can you rebase for conflicts?

**Comment by aniketpr01** [comment]
Done

File: crates/eth-rpc/src/api/eth_api.rs [hunk 1]
```diff
@@ -211,4 +211,28 @@ pub trait EthApi {
...
+    #[method(name = "newPendingTransactionFilter")]
+    async fn new_pending_transaction_filter(&self) -> Result<U64>;
+
+    /// Destroys a filter based on filter ID
+    #[method(name = "uninstallFilter")]
+    async fn uninstall_filter(&self, id: U64) -> Result<bool>;
+
+    /// Returns a list of all logs based on filter ID since the last log retrieval
+    #[method(name = "getFilterChanges")]
+    async fn get_filter_changes(&self, id: U64) -> Result<Vec<Log>>;
```
Comment (greged93) [inline crates/eth-rpc/src/api/eth_api.rs:233]: You are not using the correct return type for the function, please have a look here:
https://github.com/ethereum/execution-apis/blob/main/src/schemas/filter.yaml#L1

Same for getFilterLogs.

Please note [here](https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc-types/src/eth/filter.rs#L821) the type used by Reth.

**Review by greged93 — CHANGES_REQUESTED** [review]
One comment

File: crates/eth-rpc/src/api/eth_api.rs (hunk 1 above)
Comment (aniketpr01) [reply crates/eth-rpc/src/api/eth_api.rs]: Resolved!

File: crates/eth-rpc/src/api/eth_api.rs [hunk 2]
```diff
@@ -211,4 +211,28 @@ pub trait EthApi {
...
+    #[method(name = "uninstallFilter")]
+    async fn uninstall_filter(&self, id: U64) -> Result<bool>;
+
+    /// Returns a list of all logs based on filter ID since the last log retrieval
+    #[method(name = "getFilterChanges")]
+    async fn get_filter_changes(&self, id: U64) -> Result<Vec<FilterChanges>>;
+
+    /// Returns a list of all logs based on filter ID
+    #[method(name = "getFilterLogs")]
+    async fn get_filter_logs(&self, id: U64) -> Result<Vec<FilterChanges>>;
```
Comment (greged93) [inline crates/eth-rpc/src/api/eth_api.rs:237]: Same as above

**Review by greged93 — COMMENTED** [review]
Just remove the `Vec` and you're good to go

**Pull request merged**